go/master/client.go

```diff
@@ -104,25 +103,44 @@ func NewClient(opts ...func(*Client) error) (*Client, error) {
 		if err != nil {
 			return nil, err
 		}
-
 	}
-
+	c.passID = 0
+	c.ch = make(chan record, c.bufSize)
+	// FIXME: connection is created asyncrosly in monitorMaster go routine,
+	//        ensure the connection is ready for use before calling c.addClient.
+	time.Sleep(time.Second)
 	return c, nil
 }
 
+// StartGetRecords must be called at beginning of each pass
+func (c *Client) StartGetRecords(passID int) {
+	c.passID = passID
```

Comment:
CI failed because
```
[18:32:04] :	 [Step 1/1] WARNING: DATA RACE
[18:32:04] :	 [Step 1/1] Read at 0x00c4200782c0 by goroutine 72:
[18:32:04] :	 [Step 1/1]   github.com/PaddlePaddle/Paddle/go/master.(*Client).getRecords()
[18:32:04] :	 [Step 1/1]       /paddle/build/go/src/github.com/PaddlePaddle/Paddle/go/master/client.go:123 +0x75
[18:32:04] :	 [Step 1/1] 
[18:32:04] :	 [Step 1/1] Previous write at 0x00c4200782c0 by goroutine 7:
[18:32:04] :	 [Step 1/1]   [failed to restore the stack]
[18:32:04] :	 [Step 1/1] 
[18:32:04] :	 [Step 1/1] Goroutine 72 (running) created at:
[18:32:04] :	 [Step 1/1]   github.com/PaddlePaddle/Paddle/go/master.(*Client).StartGetRecords()
[18:32:04] :	 [Step 1/1]       /paddle/build/go/src/github.com/PaddlePaddle/Paddle/go/master/client.go:118 +0x67
[18:32:04] :	 [Step 1/1]   github.com/PaddlePaddle/Paddle/go/master.TestGetFinishTask()
[18:32:04] :	 [Step 1/1]       /paddle/build/go/src/github.com/PaddlePaddle/Paddle/go/master/client_internal_test.go:155 +0x879
[18:32:04] :	 [Step 1/1]   testing.tRunner()
[18:32:04] :	 [Step 1/1]       /usr/local/go/src/testing/testing.go:657 +0x107
```

line 117 `c.passID = passID` and line 123 `t, err := c.getTask(c.passID)` is a data race (not obvious)

It's trigged by
```go
for i := 0; i < 10; i++ {
  // init pass data                                                                                                                                                     
  c.StartGetRecords(i)
  checkOnePass(i)
}
```

It's **not** data race if only `c.StartGetRecords` is only called once, because `go` keyword is a barrier. So "line 123 `t, err := c.getTask(c.passID)`" happens after "line 117 `c.passID = passID`". But there is no guarantee if `c.StartGetRecords` is called multiple times.

Go race detector is good!

I think the easiest solution is remove `passID` from the Client struct, change it to a function parameter:
```go
func (c *Client) StartGetRecords(passID int) {
  go c.getRecords(passID)
}
```